warehouse/migrations/versions/be4cf6b58557_add_requires_python_to_release_files.py

```diff
@@ -0,0 +1,106 @@
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Add a requires_python column to release_files; pursuant to enabling PEP 503.
+
+Revision ID: be4cf6b58557
+Revises: 3d2b8a42219a
+Create Date: 2016-09-15 04:12:53.430363
+"""
+
+from alembic import op
+import sqlalchemy as sa
+
+
+revision = 'be4cf6b58557'
+down_revision = '3d2b8a42219a'
+
+
+def upgrade():
+    """
+    Add column `requires_python` in the `release_files` table.
+    """
+    op.add_column("release_files",
+                  sa.Column("requires_python",
+                            sa.Text(),
+                            nullable=True)
+                  )
+
+    # Populate the column with content from release.requires_python.
+    op.execute(
+        """ UPDATE release_files
+            SET requires_python = releases.requires_python
+            FROM releases
+            WHERE
+                release_files.name=releases.name
+                AND release_files.version=releases.version;
+        """
+    )
+
+    # Setup a trigger function to ensure that requires_python value on
+    # releases is always canonical. Avoids infinite regress by checking
+    # that the value is distinct from what it was before.
+    op.execute(
+        """CREATE OR REPLACE FUNCTION update_release_files_requires_python()
+            RETURNS TRIGGER AS $$
+            BEGIN
+                IF (TG_OP = 'INSERT'
+                        OR
+                    OLD.requires_python IS DISTINCT FROM NEW.requires_python)
+                THEN
+                UPDATE
+                    release_files
+                SET
+                    requires_python = releases.requires_python
+                FROM releases
+                WHERE
+                    release_files.name=releases.name
+                    AND release_files.version=releases.version
+                    AND release_files.name = NEW.name
+                    AND releases.version = NEW.version;
+                END IF;
+                RETURN NULL;
+            END;
+            $$ LANGUAGE plpgsql;
+        """
+    )
+
+    # Establish a trigger such that on INSERT/UPDATE on releases we update
+    # release_files with the appropriate requires_python values.
+    op.execute(
+        """ CREATE TRIGGER releases_requires_python
+            AFTER INSERT OR UPDATE ON releases
+            FOR EACH ROW
+                EXECUTE PROCEDURE update_release_files_requires_python();
+        """
+    )
+
+    # Establish a trigger such that on INSERT/UPDATE on release_files
+    # if someone changes the requires_python value, it is regenerated from
+    # releases.
+    op.execute(
+        """ CREATE TRIGGER release_files_requires_python
```

Comment:
Does that mean that those commands should be added to the `downgrade()` function? Or just that there's a clean way to address the issue if it does happen to come up.

Thanks for the feedback either way!